com.unity.shadergraph/Editor/Drawing/Views/MaterialGraphView.cs

```diff
@@ -878,9 +891,19 @@ string SerializeGraphElementsImplementation(IEnumerable<GraphElement> elements)
             var groups = elements.OfType<ShaderGroup>().Select(x => x.userData);
             var nodes = elements.OfType<IShaderNodeView>().Select(x => x.node).Where(x => x.canCopyNode);
             var edges = elements.OfType<Edge>().Select(x => (Graphing.Edge)x.userData);
-            var inputs = selection.OfType<BlackboardPropertyView>().Select(x => x.userData as ShaderInput).ToList();
             var notes = elements.OfType<StickyNote>().Select(x => x.userData);
 
+            var categories = new List<CategoryData>();
+            foreach (var selectable in selection.ToList())
```

Comment:
Why `ToList()`? Can't you just enumerate the selection?

Reply:
Removed the `ToList` here was unnecessary 👍